### PR TITLE
[ci skip] tbb: update to 2021.5.0; update reverse dependencies

### PR DESCRIPTION
**Pull request by cinerea0**
#### Testing the changes
- I tested the changes in this PR: **YES**

Reverse Dependencies:
- [x] blender
- [x] embree
- [x] occt
- [x] openimagedenoise
- [x] OpenSubdiv
- [x] openvdb
- [x] PrusaSlicer
- [x] SuiteSparse
- [x] sysdig
- [x] openimageio

Closes #32166 and closes #35105.

### CODE REVIEW

**Comment by notthewave** [comment]
I atempted some of this. Take a look at:
https://github.com/notthewave/void-packages/commits/opencolorio
Maybe that helps...

**Comment by notthewave** [comment]
I got stuck trying to make it work with musl
```
/usr/bin/ld: gnu_10.2_cxx11_64_none/libtbb.so.12.5: undefined reference to `swapcontext'
/usr/bin/ld: gnu_10.2_cxx11_64_none/libtbb.so.12.5: undefined reference to `getcontext'
/usr/bin/ld: gnu_10.2_cxx11_64_none/libtbb.so.12.5: undefined reference to `makecontext'
```

File: srcpkgs/tbb/template [hunk 1]
```diff
@@ -1,76 +1,43 @@
...
-		make_build_args="arch=ppc32"
-		;;
-	*)
-		# make sure to fall back instead of letting host uname control it
-		make_build_args="arch=generic"
-		;;
-esac
+if [ "$XBPS_TARGET_NO_ATOMIC8" ]; then
+	makedepends+=" libatomic-devel"
+	CFLAGS+=" -latomic"
```
Comment (ericonr) [inline srcpkgs/tbb/template:18]: This should theoretically be `LDFLAGS=`

File: srcpkgs/tbb/template [hunk 2]
```diff
@@ -1,76 +1,43 @@
...
+fi
 
-post_extract() {
-	sed -i 's|CPLUS = g..|CPLUS = $(CXX)|g' build/linux.gcc.inc
-	sed -i 's|CONLY = gcc|CONLY = $(CC)|g' build/linux.gcc.inc
+if [ "$XBPS_TARGET_LIBC" = "musl" ]; then
+	makedepends+=" libucontext-devel"
+fi
```
Comment (ericonr) [inline srcpkgs/tbb/template:23]: This has the unfortunate side effect of making it only really work on x86 and aarch64 :/

Idk how many users on ARM we have, though. And well, there's not much we can do.

File: srcpkgs/tbb/template [hunk 3]
```diff
@@ -1,76 +1,43 @@
...
-
 tbb-devel_package() {
 	depends="${sourcepkg}>=${version}_${revision}"
 	short_desc+=" - development files"
 	pkg_install() {
 		vmove usr/include
-		vmove "usr/lib/*.so"
 		vmove usr/lib/cmake
+		vmove usr/lib/pkgconfig
+		vmove usr/lib/*.so
```
Comment (ericonr) [inline srcpkgs/tbb/template:41]: You dropped the quotes ;)

File: srcpkgs/tbb/template (hunk 2 above)
Comment (cinerea0) [reply srcpkgs/tbb/template:23]: It's not working here either, so we don't have to worry about that yet. Are there any ways to get around the `undefined reference to 'swapcontext', 'getcontext', 'makecontext'` errors?

**Comment by cinerea0** [comment]
Because SuiteSparse took about an hour to build locally for me, the current plan is to test all of the dependent packages listed in the first comment on their own, see if they work in CI, and then push all of them at the end while skipping CI.

**Comment by cinerea0** [comment]
The good news is that I learned on IRC how to make tbb work on armv6l-musl, so there's no need to impose any extra restrictions on packages that depend on tbb. The bad news is that I now have to re-run CI for all of the reverse dependencies.

**Comment by cinerea0** [comment]
The aarch64 CI builders are clogged up with icu updates *again*, for like the fourth day in a row, so I can't make much more progress tonight.

**Comment by notthewave** [comment]
since this was reverted in cbda3de66d09d595bc2ae8ce3bfe9cee01a9d150 what needs to be done to get this merged again?

**Comment by Duncaen** [comment]
toolchain/gcc/libstdc++ needs to be updated.

**Comment by paper42** [comment]
EasyEffects uses a fuction that was removed in the new tbb release and tbb support isn't optional.

**Pull request closed without merging**